internal/benchmarks/src/cost-estimation.bench.ts

```diff
@@ -0,0 +1,139 @@
+/* eslint-disable import/no-extraneous-dependencies */
+
+import type { TransferParams } from 'fuels';
+import { ScriptTransactionRequest, Wallet } from 'fuels';
+import { launchTestNode, TestAssetId } from 'fuels/test-utils';
+import { bench } from 'vitest';
+
+import { CallTestContractFactory } from '../test/typegen/contracts';
+
+/**
+ * @group node
+ */
+describe('Cost Estimation Benchmarks', () => {
+  bench(
+    'should successfully get transaction cost estimate for a single contract call',
+    async () => {
+      using launched = await launchTestNode({
```

Comment:
Not sure if we want to have the test setup be part of the benchmarks because
- it's not the part of the code that we're benchmarking
- it might introduce flakiness due to variations in launch startup + contract deployment speed because of the unpredictability of the CI environment speed-wise, which could force us to increase the threshold of allowed variation for reasons irrelevant to the code path we're benchmarking

Perhaps we could make use of something like this:
```ts
  let contract: CallTestContract;
  let cleanup: () => void;
  beforeEach(async () => {
    const launched = await launchTestNode({
      contractsConfigs: [{ factory: CallTestContractFactory }],
    });

    cleanup = launched.cleanup;
    contract = launched.contracts[0];
  });

  afterEach(() => {
    cleanup();
  });


bench('should successfully get transaction cost estimate for a single contract call',
    async () => {
      const cost = await contract.functions
        .return_context_amount()
        .callParams({
          forward: [100, contract.provider.getBaseAssetId()],
        })
        .getTransactionCost();

      expect(cost.minFee).toBeDefined();
      expect(cost.maxFee).toBeDefined();
      expect(cost.gasPrice).toBeDefined();
      expect(cost.gasUsed).toBeDefined();
      expect(cost.gasPrice).toBeDefined();
    })
```

Reply:
Agreed @nedsalk